File: asc/semis_reports/report/summary_report_of_schools/summary_report_of_schools.js
// Copyright (c) 2016, Frappe Technologies and contributors
// For license information, please see license.txt
/* eslint-disable */
let loc_options = ["", "Urban", "Rural"];
let status_options = ["", "Functional", "Closed"];
let shift_options = ["", "Morning", "Afternoon", "Both"];
let gender_options = ["", "Boys", "Girls", "Mixed"];
frappe.query_reports["Summary Report of Schools"] = {
  filters: [
    {
      fieldname: "division",
      label: __("Division"),
      fieldtype: "Link",
      options: "Division",
    },
    {
      fieldname: "district",
      label: __("District"),
      fieldtype: "Link",
      options: "District",
    },
    {
      fieldname: "taluka",
      label: __("Taluka"),
      fieldtype: "Link",
      options: "Taluka",
    },
    {
      fieldname: "union_council",
      label: __("Union Council"),
      fieldtype: "Link",
      options: "Union Council",
    },
    {
      fieldname: "location",
      label: __("Location"),
      fieldtype: "Select",
      options: loc_options,
    },
    {
      fieldname: "gender",
      label: __("Gender"),
      fieldtype: "Select",
      options: gender_options,
    },
    {
      fieldname: "status",
      label: __("Status"),
      fieldtype: "Select",
      options: status_options,
    },
    {
      fieldname: "shift",
      label: __("Shift"),
      fieldtype: "Select",
      options: shift_options,
    },
  ],
};

File: asc/semis_reports/report/summary_report_of_schools/summary_report_of_schools.py
# Copyright (c) 2013, Frappe Technologies and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe import _


def execute(filters=None):
	columns, data = [], []
	school_gender = ["Boys","Girls","Mixed"]
	colmn = get_columns(school_gender,filters)
	columns = colmn[0]
	data = get_data(filters , colmn[1])
	return columns, data

def get_columns(gender,filters):
	columns = []
	columns.append({
		"label": _("Level"),
		"fieldtype": "Link",
		"fieldname": "level",
		"options": "Level",
		"width": 150}
	)
	case_string_scl=""
	if filters.get("gender")=="Boys":
		gender=["Boys"]
	elif filters.get("gender")=="Girls":
		gender=["Girls"]
	elif filters.get("gender")=="Mixed":
		gender=["Mixed"]
	for gndr in gender:
		columns.append(
			{
					"label": _("") + str(gndr) + " Schools" ,
					"fieldtype": "Int",
					"fieldname": str(gndr).lower() + "_schools" ,
					"width": 140
				}
		)
		case_string_scl += " Count( DISTINCT   CASE WHEN tabSchool.gender = '%s' THEN tabSchool.name ELSE NULL END), " %(str(gndr))
	columns.append(  
					{
					"label": _("Total Schools") ,
					"fieldtype": "Int",
					"fieldname": "Total Schools" ,
					"width": 140
				}
					)

	return columns, case_string_scl
    
def get_data(filters , case_string):
	conditions , group_by = get_condition(filters)
	
	temp_query= """Select tabLevel.name , %s COUNT(DISTINCT tabSchool.name)
				FROM tabSchool 
				CROSS JOIN tabLevel ON tabSchool.level = tabLevel.name  
				WHERE (tabSchool.docstatus != 2 ) %s %s ORDER BY tabLevel.list_order asc"""% (
			str(case_string), conditions, group_by)
	info = frappe.db.sql(temp_query,filters)
	return info

def get_condition(filters):
	conditions, group_by = "", "GROUP BY  tabLevel.name"
	if filters.get("division"):
		conditions += " AND  division = %(division)s"
	if filters.get("district"):
		conditions += "  AND  district = %(district)s"
	if filters.get("taluka"):
		conditions += "  AND  taluka = %(taluka)s"
	if filters.get("union_council"):
		conditions += "  AND  union_council = %(union_council)s"
	if filters.get("location"):
		conditions += "  AND  location = %(location)s"
	if filters.get("status"):
		conditions += "  AND  status = %(status)s"
	if filters.get("shift"):
		conditions += "  AND  shift = %(shift)s"
	return conditions, group_by